File: frontend/src/types/invoice.ts
export type InvoiceStatus =
  | 'draft'
  | 'active'
  | 'paid'
  | 'disputed'
  | 'escalation_ready'
  | 'escalated'
  | 'closed'

export type ReminderStage = 'gentle' | 'firm' | 'pre_escalation' | 'escalation_ready'

export interface Invoice {
  id: number
  invoice_number: string
  customer_name: string
  customer_email: string
  issue_date: string
  due_date: string
  amount_due: string
  currency: string
  status: InvoiceStatus
  current_stage: ReminderStage
  reminder_count: number
  next_action_date: string | null
  escalation_consent: boolean
  paid_at: string | null
  created_at: string
  updated_at: string
}

export interface InvoiceListItem extends Invoice {
  overdue_days: number
  interest_amount: string
}

export interface InvoiceDetailResponse {
  invoice: Invoice
  overdue_days: number
  interest_amount: string
}

export interface TimelineItem {
  id: number
  stage: ReminderStage
  channel: string
  recipient: string
  subject: string
  body: string
  action_type: string
  sent_at: string
}

File: backend/app/schemas/invoice.py
"""Pydantic schemas for invoice and reminder timeline records."""
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, EmailStr, Field

from app.models.enums import InvoiceStatus, ReminderStage


class InvoiceBase(BaseModel):
    invoice_number: str = Field(min_length=1, max_length=64)
    customer_name: str = Field(min_length=1, max_length=255)
    customer_email: EmailStr
    customer_phone: str | None = None
    issue_date: date
    due_date: date
    amount_due: Decimal = Field(gt=0)
    currency: str = Field(default="INR", min_length=3, max_length=8)
    notes: str | None = None


class InvoiceCreate(InvoiceBase):
    escalation_consent: bool = False


class InvoiceRead(InvoiceBase):
    id: int
    status: InvoiceStatus
    current_stage: ReminderStage
    reminder_count: int
    next_action_date: date | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ReminderLogRead(BaseModel):
    id: int
    invoice_id: int
    stage: ReminderStage
    channel: str
    recipient: str
    subject: str
    body: str
    action_type: str
    sent_at: datetime

    class Config:
        from_attributes = True
